commit 2198c2c15eeeefe5369d7fa56a9d42a19950d4fb
Author: Jernej Skrabec <jernej.skrabec@siol.net>
Date:   Sun Nov 4 19:27:00 2018 +0100

    drm/sun4i: Initialize registers in tcon-top driver
    
    [ Upstream commit c96d62215fb540e2ae61de44cb7caf4db50958e3 ]
    
    It turns out that TCON TOP registers in H6 SoC have non-zero reset
    value. This may cause issues if bits are not changed during
    configuration.
    
    To prevent that, initialize registers to 0.
    
    Signed-off-by: Jernej Skrabec <jernej.skrabec@siol.net>
    Signed-off-by: Maxime Ripard <maxime.ripard@bootlin.com>
    Link: https://patchwork.freedesktop.org/patch/msgid/20181104182705.18047-24-jernej.skrabec@siol.net
    Signed-off-by: Sasha Levin <sashal@kernel.org>

diff --git a/drivers/gpu/drm/sun4i/sun8i_tcon_top.c b/drivers/gpu/drm/sun4i/sun8i_tcon_top.c
index d5240b777a8f..adcdf946c365 100644
--- a/drivers/gpu/drm/sun4i/sun8i_tcon_top.c
+++ b/drivers/gpu/drm/sun4i/sun8i_tcon_top.c
@@ -168,6 +168,13 @@ static int sun8i_tcon_top_bind(struct device *dev, struct device *master,
 		goto err_assert_reset;
 	}
 
+	/*
+	 * At least on H6, some registers have some bits set by default
+	 * which may cause issues. Clear them here.
+	 */
+	writel(0, regs + TCON_TOP_PORT_SEL_REG);
+	writel(0, regs + TCON_TOP_GATE_SRC_REG);
+
 	/*
 	 * TCON TOP has two muxes, which select parent clock for each TCON TV
 	 * channel clock. Parent could be either TCON TV or TVE clock. For now
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1007_2198c2c15eeeefe5369d7fa56a9d42a19950d4fb.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1007_2198c2c15eeeefe5369d7fa56a9d42a19950d4fb.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
